packages/core/test/errors.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const errors = require('../src/errors');
+const createAppRequestClient = require('../src/tools/create-app-request-client');
+const createInput = require('../src/tools/create-input');
+
+describe('errors', () => {
+  it('should name errors', () => {
+    const error = new errors.CheckError('Check Mate!');
+
+    error.should.instanceOf(errors.CheckError);
+    error.name.should.eql('CheckError');
+    error.message.should.eql('Check Mate!');
+  });
+
+  describe('AppError', () => {
+    it('should stringify arguments', () => {
+      const error = new errors.Error('My Message', 'MyCode', 400);
+
+      error.should.instanceOf(errors.Error);
+      error.name.should.eql('AppError');
+      error.message.should.eql(
+        '{"message":"My Message","code":"MyCode","status":400}'
+      );
+    });
+  });
+
+  describe('ResponseError', () => {
+    it('should stringify arguments', async () => {
+      const testLogger = () => Promise.resolve({});
+      const input = createInput({}, {}, testLogger);
+      const request = createAppRequestClient(input);
+
+      const response = await request({ url: 'https://httpbin.org/status/400' });
```

Comment:
We should use our in-house httpbin `https://httpbin.zapier-tooling.com/status/400` here.

Reply:
Fine with me, but https://github.com/zapier/zapier-platform/blob/3a3fd6a43160e6bee36b919dd9ded1c1baf57e3b/packages/core/test/http-middleware.js#L349 is the only test that uses that at the moment and all other tests use `https://httpbin.org/`, so if we prefer our in-house I suggest we switch all tests in a separate PR.

Reply:
> we prefer our in-house I suggest we switch all tests in a separate PR

Sounds good!

Reply:
Ours behaves slightly differently (iirc wraps everything in arrays when the public one doesn't), so it's just enough friction that we haven't bothered to update it yet.

Reply:
Looks like y'all are 🆗 with swapping all in a separate PR.